### PR TITLE
Add PreviousExposure Screen

### PR DESCRIPTION
**Pull request by selvach**
Adds a new Previous Exposure Screen to the Profiling Flow. 
- Moves a few questions from the AboutYourHealth screen to the PreviousExposureScreen
- Adds 4 questions:
    - unusual_symptoms_month_before: boolean,
    - still_experiencing_unusual_symptoms: boolean,
    - unusual_symptoms_days_ago: number,
    - unusual_symptoms_changed: boolean,

### CODE REVIEW

File: src/features/patient/PreviousExposure.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            {label: 'Much better', value: 'much_better'},
+            {label: 'A little better', value: 'little_better'},
+            {label: 'The same', value: 'same'},
+            {label: 'A little worse', value: 'little_worse'},
+            {label: 'Much worse', value: 'much_worse'},
+        ]
+        return (
+            <Screen profile={currentPatient.profile}>
+                <Header>
+                    <HeaderText>Previous Exposure to COVID</HeaderText>
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:146]: Need to use i18n

File: src/features/patient/PreviousExposure.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    render() {
+        const currentPatient = this.props.route.params.currentPatient;
+        const symptomChangeChoices = [
+            {label: 'Much better', value: 'much_better'},
+            {label: 'A little better', value: 'little_better'},
+            {label: 'The same', value: 'same'},
+            {label: 'A little worse', value: 'little_worse'},
+            {label: 'Much worse', value: 'much_worse'},
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:141]: Needs i18n for the labels

File: src/features/patient/PreviousExposure.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                >
+                    {props => {
+                        return (
+                            <KeyboardAvoidingView behavior={Platform.OS === "ios" ? "padding" : undefined}>
+                                <Form>
+
+                                    <DropdownField
+                                        selectedValue={props.values.unusualSymptomsMonthBefore}
+                                        onValueChange={props.handleChange("unusualSymptomsMonthBefore")}
+                                        label={"Do you have unusual symptoms in the month before you started reporting on this app?"}
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:168]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                                    />
+
+                                    {props.values.unusualSymptomsMonthBefore === 'yes' && (
+                                        <DropdownField
+                                            selectedValue={props.values.stillExperiencingUnusualSymptoms}
+                                            onValueChange={props.handleChange("stillExperiencingUnusualSymptoms")}
+                                            label={"Are you still experiencing symptoms?"}
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:175]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                                        />
+                                    )}
+
+                                    {props.values.stillExperiencingUnusualSymptoms === 'yes' && (
+                                        <>
+                                            <GenericTextField
+                                                formikProps={props}
+                                                label="How many days ago did your symptoms start?"
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:183]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+                                                name="unusualSymptomsDaysAgo"
+                                                keyboardType="numeric"
+                                            />
+                                            <DropdownField
+                                                selectedValue={props.values.unusualSymptomsChanged}
+                                                onValueChange={props.handleChange("unusualSymptomsChanged")}
+                                                label={"How have your symptoms changed over the last few days?"}
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:190]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+                                                items={symptomChangeChoices}
+                                            />
+                                        </>
+                                    )}
+
+                                    {props.values.unusualSymptomsMonthBefore === 'yes' && (
+                                        <DropdownField
+                                            selectedValue={props.values.alreadyHadCovid}
+                                            onValueChange={props.handleChange("alreadyHadCovid")}
+                                            label={"Do you think you have already had COVID -19, but were not tested)?"}
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:200]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+                                        />
+                                    )}
+
+                                    {props.values.alreadyHadCovid === 'yes' && (
+                                        <DropdownField
+                                            selectedValue={props.values.classicSymptoms}
+                                            onValueChange={props.handleChange("classicSymptoms")}
+                                            label={"Did you have the classic symptoms (high fever and persistent cough) for several days?"}
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:208]: i18n

File: src/features/patient/PreviousExposure.tsx [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+
+        const userService = new UserService();
+        let infos = this.createPatientInfos(formData);
+
+        userService.updatePatient(patientId, infos)
+            .then(response => {
+                this.props.navigation.navigate('StartAssessment', {currentPatient});
+            })
+            .catch(err => {
+                this.setState({errorMessage: "Something went wrong, please try again later"})
```
Comment (the-kenneth) [inline src/features/patient/PreviousExposure.tsx:99]: i18n

**Comment by selvach** [comment]
Pushed a chance base on Abu's request to rename a backend field for consistency. Did another check end to end.

**Comment by selvach** [comment]
It's all been done!

File: src/core/user/dto/UserAPIContracts.ts [hunk 10]
```diff
@@ -68,10 +68,30 @@ export type PatientInfosRequest = {
...
+    past_symptom_chest_pain: boolean,
+    past_symptom_hoarse_voice: boolean,
+    past_symptom_abdominal_pain: boolean,
+    past_symptom_delirium: boolean,
+
     already_had_covid: boolean,
     classic_symptoms: boolean,
     classic_symptoms_days_ago: number,
 
+    //
```
Comment (the-kenneth) [inline src/core/user/dto/UserAPIContracts.ts:94]: Unused comment

Comment (selvach) [reply src/core/user/dto/UserAPIContracts.ts]: Thanks! Fixed this and re-grouped.

File: src/core/user/dto/UserAPIContracts.ts [hunk 11]
```diff
@@ -39,7 +36,7 @@ export type PatientInfosRequest = {
     same_household_as_reporter: boolean,
 
     year_of_birth: number,
-    gender: number, // 0: female, 1: male
+    gender: number, // 0: female, 1: male, 2: pfnts 3: pfnts
```
Comment (abubuwe) [inline src/core/user/dto/UserAPIContracts.ts:39]: 3 is Intersex not pfnts

Comment (selvach) [reply src/core/user/dto/UserAPIContracts.ts:39]: Oh oh....

            gender: formData.sex === 'male' ? 1 : formData.sex === 'female' ? 0 : formData.sex === 'pfnts' ? 2 : 3,

Comment (selvach) [reply src/core/user/dto/UserAPIContracts.ts:39]: This is a new bug. Can we solve in a different PR, please.

**Pull request merged**